### PR TITLE
Added top level syntax & semantics ToC

### PR DESCRIPTION
**Pull request by codyjb**
Fixes https://github.com/crystal-lang/crystal-book/issues/213

### CODE REVIEW

**Comment by RX14** [comment]
If we do this, shouldn't we do this for all sections?

**Comment by codyjb** [comment]
Yeah, I think so, this was just the most egregious offender.

File: syntax_and_semantics/README.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
 # Syntax and Semantics
 
 The program’s source code must be encoded in UTF-8.
+
+* [Comments](comments.html)
+* [Literals](literals.html)
+* [Assignment](assignment.html)
+* [Local Variabls](local_variables.html)
```
Comment (mverzilli) [inline syntax_and_semantics/README.md]: Local Variables

**Comment by mverzilli** [comment]
I think it's alright if we add the TOC pages little by little, or at least I'd rather have a few of them than none just for the sake of consistency.

**Pull request merged**